File: java/core/src/main/java/com/github/fernthedev/lightchat/core/api/APIUsage.java
package com.github.fernthedev.lightchat.core.api;

import java.lang.annotation.*;

/**
 * Just to notify Intellij that usage will not be required at compile-time because it's an API method
 */
@Retention(RetentionPolicy.SOURCE)
@Target({ElementType.ANNOTATION_TYPE, ElementType.TYPE, ElementType.METHOD, ElementType.FIELD, ElementType.PARAMETER})
@APIUsage
@Documented
public @interface APIUsage { }

File: java/core/src/main/java/com/github/fernthedev/lightchat/core/api/APIUsage.kt
package com.github.fernthedev.lightchat.core.api

/**
 * Just to notify Intellij that usage will not be required at compile-time because it's an API method
 */
@Retention(AnnotationRetention.SOURCE)
@Target(
    AnnotationTarget.ANNOTATION_CLASS,
    AnnotationTarget.CLASS,
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.FIELD,
    AnnotationTarget.VALUE_PARAMETER
)
@APIUsage
@MustBeDocumented
annotation class APIUsage 